### ISSUE TITLE
cmd/go: go mod tidy removes too much when modules are used for multiple packages

### ISSUE THREAD

**Issue by integrii** [golang/go#31059]
### What version of Go are you using (`go version`)?

<pre>
go version go1.12.1 darwin/amd64

</pre>

### Does this issue reproduce with the latest release?

yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
GOARCH="amd64"
GOBIN="/Users/egreer200/go/bin"
GOCACHE="/Users/egreer200/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/egreer200/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/Cellar/go/1.12.1/libexec"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.12.1/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/7h/qd9xrbq10lb03g4lkqcl_cdr0000gp/T/go-build194066640=/tmp/go-build -gno-record-gcc-switches -fno-common"
</pre></details>

### What did you do?

I have a module that has a couple layers of packages under it.  The module is [here](https://github.com/Comcast/kuberhealthy/tree/master/pkg).  Specifically notice that the `checks` directory has other packages within it.

I can `go mod tidy` from the `pkg/` directory linked above.

### What did you expect to see?

A few packages that were not used possibly being removed.

### What did you see instead?

Nearly all packages were removed, but some were kept.  It looked like this:

![image](https://user-images.githubusercontent.com/98695/55023970-21705780-4fbb-11e9-9277-ad5293565e55.png)
![image](https://user-images.githubusercontent.com/98695/55023984-27fecf00-4fbb-11e9-9080-660872c112d4.png)
and so on.

It appears that tidy is targeting only one (or a few) of the packages.  Perhaps only the packages at the first level of directory depth under the `pkg/` dir were analyzed? 

In short, it looks like `go mod tidy` will not work when a module is used for multiple packages.

**Comment by integrii** [comment]
Actually, I think this is because when I created this module, I ran `go build ...` and it started building literally everything in my gopath...  I am opening a new issue.  I believe `go mod tidy` is actually correct.  After running a tidy, I can run `go build ./...` and the project builds fine!
